### ISSUE TITLE
test(model): Qwen35 Vulkan MTP raw-hidden commit history is bit-flaky in continuation logits

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13145]
# test(model): Qwen35 Vulkan MTP raw-hidden commit history is bit-flaky in continuation logits

```routing
lane: model
paths: internal/model/qwen35_mtp_target_test.go
expected_steps: 2
```

### Work unit
leaf

### Generation
gen/now

### Lane
internal/model

## Value

- Centrality: Peripheral
- P1 Context: preserved - the test compares a bit-exact continuation-logit oracle; no runtime context is duplicated.
- P2 Net value: preserved - removes recurring false-red from the model test lane so real regressions are not masked.
- P3 Adaptation: N/A - pure test determinism, no adaptive behavior.
- P4 Operations: preserved - existing operator path and the Vulkan MTP route are unchanged.

## Parent context

#12235 (epic: end-to-end Apple Silicon Metal MTP speculative decoding).

## Current state

`TestQwen35VulkanMTPRawHiddenCommitHistory` fails intermittently on clean trunk
with a last-ULP mismatch in the continuation logits, for example:

```
continuation logits[0] = bebb4728, want bebb472c (-0.36577725 vs -0.36577737)
continuation logits[0] = 3ee3695e, want 3ee3695c (0.44416326 vs 0.4441632)
```

The test asserts `assertFloat32BitsEqual(t, "continuation logits", gotLogits, wantLogits)`
between the transaction target and a fresh oracle session. The difference is a
single float32 ULP, consistent with a reduction-order / instruction-selection
sensitivity rather than a logic defect. Reproduced on clean trunk (`fc34b6cc5`)
with no local changes.

## Why this is next

The Metal MTP admission work re-ran the model suite and re-hit this flake. It is
independent of that change (fails identically on a stashed clean tree), so it
must be fixed or quarantined before it can mask a genuine MTP regression.

## Working spine

Reproduce the ULP mismatch deterministically -> determine whether the
reduction order differs between the transaction target and the oracle path ->
either make both paths share the reducing kernel or replace the bit-exact
assertion with a bounded-ULP tolerance plus an explicit rationale -> confirm the
test is stable across repeated runs.

## Core through-line

Run `TestQwen35VulkanMTPRawHiddenCommitHistory` repeatedly -> observe the ULP
delta -> localize the differing reduction/instruction path -> apply the narrow
fix (shared kernel or documented tolerance) -> assert stability over N runs.

## Gold-plating boundary

Do not weaken any other bit-exact MTP parity assertion, do not change model
math, and do not introduce tolerance where exactness is the actual contract.

## Scope

In scope: the flaky bit-exact continuation-logit assertion in
`internal/model/qwen35_mtp_target_test.go`.

## Definition of done

- [ ] The ULP mismatch is localized to a concrete differing reduction/instruction path, or bounded with an explicit documented tolerance.
- [ ] `go test ./internal/model/ -run 'TestQwen35VulkanMTPRawHiddenCommitHistory' -count=20` is green.
- [ ] No other bit-exact MTP parity assertion is weakened.

## Done condition

`go test ./internal/model/ -run 'TestQwen35VulkanMTPRawHiddenCommitHistory' -count=20`
is green with no ULP mismatch.

## Witness

```bash
go test ./internal/model/ -run 'TestQwen35VulkanMTPRawHiddenCommitHistory' -count=20 -timeout=600s
```

### Verifiable Witness

```bash
go test ./internal/model/ -run 'TestQwen35VulkanMTPRawHiddenCommitHistory' -count=20 -timeout=600s
```

### File:Line Seam

`internal/model/qwen35_mtp_target_test.go:135`

### Blast Radius & Affected Lanes

- **Primary Lane:** `internal/model` (test only)
- **Unaffected Lanes:** production MTP/Vulkan paths untouched.

### Quarantined Fallback Mechanism

If the two paths legitimately cannot share a reduction kernel, bound the
assertion with an explicit ULP tolerance and document why exact bit equality is
not attainable, rather than deleting the check.

## Stop conditions

Stop and retain a `t.Skip` with the exact reproducer if the mismatch cannot be
localized within this leaf; never delete the assertion silently.

## Rollback / downgrade

Revert the assertion change to restore the current (flaky) behavior.

## Acceptance gate

`go test ./internal/model/ -run 'TestQwen35VulkanMTPRawHiddenCommitHistory' -count=20 -timeout=600s` green.

## Closure binding

Resolving commit cites the issue and carries `(fak model)`.

## Work estimate

Estimate: 2 points.

## Overall completion contribution

Parent scope baseline: #12235, 20 points. Contribution: 2/20 points.

## Target operating envelope

- test_runs: >= 20 runs
- ulp_mismatches: 0 mismatches

## Witnessed envelope

- test_runs: 20 runs
- ulp_mismatches: 0 mismatches

## Bounded write set

- `internal/model/qwen35_mtp_target_test.go`

## Completion standard
production
